pallets/nfts/src/lib.rs

```diff
@@ -233,25 +262,54 @@ impl<T: Config> NFTs for Pallet<T> {
     type AccountId = T::AccountId;
     type NFTDetails = T::NFTDetails;
     type NFTId = T::NFTId;
+    type NFTSeriesId = T::NFTSeriesId;
 
     fn create(
         owner: &Self::AccountId,
         details: Self::NFTDetails,
+        series_id: Self::NFTSeriesId,
     ) -> result::Result<Self::NFTId, DispatchError> {
-        let id = Total::<T>::get();
-        Total::<T>::put(id.checked_add(&1.into()).ok_or(Error::<T>::NFTIdOverflow)?);
+        let nft_id = Total::<T>::get();
+        Total::<T>::put(
+            nft_id
+                .checked_add(&1.into())
+                .ok_or(Error::<T>::NFTIdOverflow)?,
+        );
+
+        let series_id = if series_id == Default::default() {
+            None
+        } else {
+            Some(series_id)
+        };
+
         Data::<T>::insert(
-            id,
+            nft_id,
             NFTData {
                 owner: owner.clone(),
                 details,
                 sealed: false,
                 locked: false,
+                series_id: series_id.clone(),
             },
         );
 
-        Self::deposit_event(Event::Created(id, owner.clone()));
-        Ok(id)
+        if let Some(series_id) = series_id {
+            if Series::<T>::contains_key(series_id) {
+                Series::<T>::mutate(series_id, |vec| vec.push(nft_id));
```

Comment:
This would produce one extra, unnecessary read. You can try `mutate_exists` which will allow you to detect wether the stuff already exists in storage

Reply:
Changed it to mutate_exists.